src/shared/cards/Card Template.js

```diff
@@ -0,0 +1,76 @@
+import Card from 'cards/Card';
+
+export default class CardName extends Card {
+  static title = 'Card Name';
+  cost = 4;
+  VP = 3;
+  coin = 2;
+  types = new Set(['Action', 'Treasure', 'Attack', 'Reaction', 'Victory']);
+  async onPlay(player) {
+    // Vanilla Effects
+    player.draw(1);
+    player.actions++;
+    player.buys++;
+    player.money++;
+
+    // Trash card(s) in Hand, is it game.currentPlayer, or just Player?
+    const cards = await game.currentPlayer.selectCards(min, max, predicate);
+    cards.forEach(card => {
+      player.trash(card);
+    })
+
+    // Trash this card
+    player.trash(this);
+
+    // Draw card and do something if()
+    card = player.deck.last();
+    if (card.types.has('Victory')) {  // look at top card in deck, discard if it is a victory card
+      player.moveCard(this, player.deck, player.discard);
+    }
+
+    // Draw multiple cards and do something if(), basically emulating library
+    const cards = player.setAside(min, max);
```

Comment:
I made `player.draw` return the cards that you draw
I also made `player.draw` take a parameter that lets you decide where the card goes
eg
```
while (player.hand.size < 7) {
  await player.draw(1, async card => {
    if (card.types.has('Action') {
      const choice = await player.selectOption(['Put in hand', 'Set aside']);
      return ['hand', 'aside'][choice];
    }
    return 'hand';
  });
}
```